From 2a9644266370e1a83d787d9d33b577bf2a1fb604 Mon Sep 17 00:00:00 2001
From: Rene Engelhard <rene@debian.org>
Date: Fri, 30 Aug 2019 20:16:47 +0200
Subject: [PATCH] set -Wl,--no-as-needed for -latomic

... since gcc 9 defaults to --as-needed and that breaks the same way as
-latomic wasn't set.
(Yes, it also sets -Wl,--as-needed maybe for everything but this is collateral
damage on "obsolete" architectures.)

Change-Id: Ia60d422130a0113eb98cd476ac7f9ded2453b33d
---
 solenv/gbuild/platform/LINUX_ARM_GCC.mk     | 2 +-
 solenv/gbuild/platform/LINUX_GODSON_GCC.mk  | 2 +-
 solenv/gbuild/platform/LINUX_M68K_GCC.mk    | 2 +-
 solenv/gbuild/platform/LINUX_POWERPC_GCC.mk | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/solenv/gbuild/platform/LINUX_ARM_GCC.mk b/solenv/gbuild/platform/LINUX_ARM_GCC.mk
index e52cf2bb8950..03870068de51 100644
--- a/solenv/gbuild/platform/LINUX_ARM_GCC.mk
+++ b/solenv/gbuild/platform/LINUX_ARM_GCC.mk
@@ -13,6 +13,6 @@ gb_COMPILEROPTFLAGS := -Os
 
 include $(GBUILDDIR)/platform/linux.mk
 
-gb_LinkTarget_LDFLAGS += -latomic
+gb_LinkTarget_LDFLAGS += -Wl,-no-as-needed -latomic -Wl,--as-needed
 
 # vim: set noet sw=4:
diff --git a/solenv/gbuild/platform/LINUX_GODSON_GCC.mk b/solenv/gbuild/platform/LINUX_GODSON_GCC.mk
index f16f98530e2b..753747ece789 100644
--- a/solenv/gbuild/platform/LINUX_GODSON_GCC.mk
+++ b/solenv/gbuild/platform/LINUX_GODSON_GCC.mk
@@ -13,6 +13,6 @@ gb_COMPILEROPTFLAGS := -Os
 
 include $(GBUILDDIR)/platform/linux.mk
 
-gb_LinkTarget_LDFLAGS += -latomic
+gb_LinkTarget_LDFLAGS += -Wl,-no-as-needed -latomic -Wl,--as-needed
 
 # vim: set noet sw=4:
diff --git a/solenv/gbuild/platform/LINUX_M68K_GCC.mk b/solenv/gbuild/platform/LINUX_M68K_GCC.mk
index 942696f80231..8e5e34c4a6e8 100644
--- a/solenv/gbuild/platform/LINUX_M68K_GCC.mk
+++ b/solenv/gbuild/platform/LINUX_M68K_GCC.mk
@@ -12,6 +12,6 @@ gb_COMPILEROPTFLAGS := -Os
 
 include $(GBUILDDIR)/platform/linux.mk
 
-gb_LinkTarget_LDFLAGS += -latomic
+gb_LinkTarget_LDFLAGS += -Wl,-no-as-needed -latomic -Wl,--as-needed
 
 # vim: set noet sw=4:
diff --git a/solenv/gbuild/platform/LINUX_POWERPC_GCC.mk b/solenv/gbuild/platform/LINUX_POWERPC_GCC.mk
index 48fa395cfada..d026cb64cb8a 100644
--- a/solenv/gbuild/platform/LINUX_POWERPC_GCC.mk
+++ b/solenv/gbuild/platform/LINUX_POWERPC_GCC.mk
@@ -12,6 +12,6 @@ gb_CPUDEFS += -DPPC
 
 include $(GBUILDDIR)/platform/linux.mk
 
-gb_LinkTarget_LDFLAGS += -latomic
+gb_LinkTarget_LDFLAGS += -Wl,--no-as-needed -latomic -Wl,--as-needed
 
 # vim: set noet sw=4:
-- 
2.20.1

